product_barcode_required/models/product.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Camptocamp SA
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl)
+
+from odoo import _, api, exceptions, models
+
+
+class BarcodeRequiredMixin(models.AbstractModel):
```

Comment:
I tend to prefer having very small models in the same file, but I've been told a few times the OCA code should respect the 1model/1class/1file.